src/Sentry.Unity.Editor/AutoInstrumentation/ModuleReadWriter.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Reflection;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace Sentry.Unity.Editor
+{
+    static class ModuleReaderWriter
+    {
+        public static (ModuleDefinition module, bool hasSymbols) Read(string file)
+        {
+            try
+            {
+                return InnerRead(file);
+            }
+            catch (Exception exception)
+            {
+                throw new($"Failed to read: {file}", exception);
+            }
+        }
+
+        private static (ModuleDefinition module, bool hasSymbols) InnerRead(string file)
+        {
+            var parameters = new ReaderParameters
+            {
+                InMemory = true,
+            };
+
+            var module = ModuleDefinition.ReadModule(file, parameters);
+            var hasSymbols = TryReadSymbols(module);
+
+            return (module, hasSymbols);
+        }
+
+        private static bool TryReadSymbols(this ModuleDefinition module)
+        {
+            var hasSymbols = false;
+            try
+            {
+                module.ReadSymbols();
+                hasSymbols = true;
+            }
+            catch (SymbolsNotFoundException)
+            {
+            }
+
+            return hasSymbols;
```

Comment:
```suggestion
            try
            {
                module.ReadSymbols();
                return true;
            }
            catch
            {
                return false;
            }
```

... unless you want other possible exceptions to bubble up?

Reply:
and would you want to log anything in the catch?